actors/miner/src/lib.rs

```diff
@@ -5427,18 +5491,20 @@ fn activate_sectors_pieces(
 /// Deals and claims are grouped by sectors
 /// Successfully activated sectors have their DealSpaces returned
 /// Failure to claim datacap for any verified deal results in the whole batch failing
+#[allow(clippy::complexity)]
 fn activate_sectors_deals(
     rt: &impl Runtime,
     activation_infos: &[DealsActivationInput],
     compute_unsealed_cid: bool,
-) -> Result<(BatchReturn, Vec<DataActivationOutput>), ActorError> {
+) -> Result<(BatchReturn, Vec<DataActivationOutput>, Vec<Vec<(Cid, u64)>>), ActorError> {
```

Comment:
The point of this DataActivationOutput type is to carry whatever per-sector activation metadata is needed by other processes. We should include the piece information in there rather than need to accumulate it in a separate map and look it up again later. That will remove some hard-to-mentally-verify indexing and unwraps.